spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -500,12 +501,12 @@ private void removeLockKey() {
 					return;
 				}
 				else if (Boolean.FALSE.equals(unlinkResult)) {
-					throw new IllegalStateException("Lock was released in the store due to expiration. " +
+					throw new ConcurrentModificationException("Lock was released in the store due to expiration. " +
```

Comment:
This is breaking change in the behavior.
We will have to mention this in the Migration Guide when we done with this PR: https://github.com/spring-projects/spring-integration/wiki